router/tests/common/mod.rs

```diff
@@ -30,18 +29,79 @@ pub const TEST_TOPIC_ID: i64 = 1;
 pub const TEST_QUERY_POOL_ID: i64 = 1;
 
 /// Common retention period value we'll use in tests
-pub const TEST_RETENTION_PERIOD_NS: Option<i64> = Some(3_600 * 1_000_000_000);
+pub const TEST_RETENTION_PERIOD_NS: i64 = 3_600 * 1_000_000_000;
+
+pub fn test_context() -> TestContextBuilder {
+    TestContextBuilder::default()
+}
+
+#[derive(Debug, Default)]
+pub struct TestContextBuilder {
+    namespace_autocreate_policy: Option<NamespaceAutocreatePolicy>,
+}
+
+impl TestContextBuilder {
+    pub fn autocreate_namespace(mut self, enabled: bool) -> Self {
```

Comment:
I don't think it is entirely standard across the rust ecosystem but sometimes such functions are prefixed with `with_` -- like

```suggestion
    pub fn with_autocreate_namespace(mut self, enabled: bool) -> Self {
```

Reply:
Makes sense, will update!